parser/src/test/scala/hmda/parser/fi/ts/TsDatParserSpec.scala

```diff
@@ -19,34 +21,43 @@ class TsDatParserSpec extends FlatSpec with MustMatchers {
     respondent.zipCode mustBe "22102"
   }
 
-  it should "parse other info about the respondent" in {
+  property("generate proper dat format") {
+    forAll(tsGen) { (ts: TransmittalSheet) =>
+      val parsedTs = TsDatParser(ts.toDAT)
+      val updatedRespondent = parsedTs.respondent.copy(id = parsedTs.respondent.id.replaceFirst("^0+(?!$)", ""))
```

Comment:
The spec states that the respondent ID should be right-justified and zero-filled, which is what the `toDAT` method outputs.  However, the parser only trims leading and trailing spaces, so we have to manually trim the extra zeroes so that the two transmittal sheets match.

We had to do the same thing on a couple of the fields in the `LarDatParserSpec`.